vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.html

```diff
@@ -97,11 +113,89 @@
+			link.css('color', 'rgba(128,128,128,1)');
+			link.css('font-size', "12px");
+			var inputRotation = document.createElement("input");
+			inputRotation.setAttribute("type", "hidden");
+			inputRotation.setAttribute("id", "spaceLinkRotation-[(${link.link.id})]");
+			inputRotation.setAttribute("value", "[(${link.rotation})]");
+			var inputLabel = document.createElement("input");
+			inputLabel.setAttribute("type", "hidden");
+			inputLabel.setAttribute("id", "spaceLinkLabel-[(${link.link.id})]");
+			inputLabel.setAttribute("value", "[(${link.link.name})]");
+			var inputType = document.createElement("input");
+			inputType.setAttribute("type", "hidden");
+			inputType.setAttribute("id", "spaceLinkType-[(${link.link.id})]");
+			inputType.setAttribute("value", "[(${link.type})]");
+			var inputLinkedSpace = document.createElement("input");
+			inputLinkedSpace.setAttribute("type", "hidden");
+			inputLinkedSpace.setAttribute("id", "spaceLinkTarget-[(${link.link.id})]");
+			inputLinkedSpace.setAttribute("value", "[(${link?.link?.targetSpace?.id})]");
+			var unpublishedSpaceElement;
+			    [# th:if="${#strings.equals(link?.link?.targetSpace?.spaceStatus,'UNPUBLISHED')}"]
+				unpublishedSpaceElement=$('<i data-link-id="main-unpublished-[(${link.link.id})]" class="fa fa-exclamation-triangle fa-lg unpublishedSpaceClass unpublishedClass-[(${link.link.id})]" style="color: var(--secondary);"></i>')
+				unpublishedSpaceElement.css('position', 'absolute');
+				unpublishedSpaceElement.css('left', [[${link.positionX}]] + posX + 25);
+				unpublishedSpaceElement.css('top', [[${link.positionY}]] + posY - 13);
+				unpublishedSpaceElement.css('transform', 'rotate([(${link.rotation})]deg)');
+				unpublishedSpaceElement.css('font-size', "12px");
+				[/]
+			//append to form element that you want .
+			
+			$("#main-space").append(inputRotation);
+			$("#main-space").append(inputLabel);
+			$("#main-space").append(inputType);
+			$("#main-space").append(inputLinkedSpace);
+			$("#main-space").append(link);
+			$("#main-space").append(unpublishedSpaceElement);
+			
+			
+			$(".slabel-[(${link.link.id})]").css({
+				'transform': 'rotate(0deg)',
+				'left': [[${link.positionX}]] + posX - 10,
+				'top': [[${link.positionY}]] + posY + 35,
+				'color': 'red'
+			});		
+			
+			$('[data-link-id="main-[(${link.link.id})]"]').css('cursor', 'pointer');
+			$('[data-link-id="main-[(${link.link.id})]"]').click(function(e) {
+			    var rotation = document.getElementById("spaceLinkRotation-[(${link.link.id})]").value;
+			    var label = document.getElementById("spaceLinkLabel-[(${link.link.id})]").value;
+			    var type = document.getElementById("spaceLinkType-[(${link.link.id})]").value;
+			    var targetSpace = document.getElementById("spaceLinkTarget-[(${link.link.id})]").value;
+				makeSpaceLinksEditable(label, [[${link.link.id}]], rotation ,targetSpace,[[${link.positionX}]],[[${link.positionY}]],[[${link.id}]],type);
+			});

```

Comment:
since we're using jquery, let's use that consistently (the whole block mixes  jquery and javascript.